src/checkers/inference/InferenceTreeAnnotator.java

```diff
@@ -114,13 +117,23 @@ public Void visitClass(final ClassTree classTree, final AnnotatedTypeMirror clas
         InferenceUtil.testArgument(classType instanceof AnnotatedDeclaredType,
                 "Unexpected type for ClassTree ( " + classTree + " ) AnnotatedTypeMirror ( " + classType + " ) ");
 
-        // For anonymous classes, we do not create additional variables, as they
-        // were already handled by the visitNewClass. This would otherwise result
-        // in new variables for an extends clause, which then cannot be inserted.
-        if (!InferenceUtil.isAnonymousClass(classTree)) {
-            this.variableAnnotator.visit(classType, classTree);
+        // Annotate the enclosing type if it exists
+        AnnotatedDeclaredType declType = (AnnotatedDeclaredType) classType;
```

Comment:
declType and enclosingType aren't used outside of `if (classPath != null)`. Can they be moved into that branch?